weapons/projectile.py

```diff
@@ -1,16 +1,35 @@
-import os
+import weakref
 
-import pygame
+from scene import state
+from utils.collision import CollisionEffect
+from utils.sprite import Sprite, Collider
 
-from utils.fg_element import FGElement
 
-class Projectile(FGElement):
-    def __init__(self, *args, **kwargs):
+class Projectile(Sprite):
+    def __init__(self, owner, *args, **kwargs):
         self.momentum_x = kwargs.pop('momentum_x', 0)
         self.momentum_y = kwargs.pop('momentum_y', 0)
+        self.owner = weakref.ref(owner)
+        self.impacted = False
 
         super().__init__(*args, **kwargs)
 
+    def set_collider(self):
+        self.collider = Collider(self, effect=CollisionEffect.Destroy, reaction=self.collided)
+
+    def collided(self, collider, effect):
+        if effect == CollisionEffect.Halt:
+            self.owner().projectile_hit()
+            self.impacted = True
+
     def update(self, model, lag_scalar):
-        self.x = self.x + (self.move_x + self.momentum_x) * lag_scalar
-        self.y = self.y + (self.move_y + self.momentum_y) * lag_scalar
+        super().update(model, lag_scalar)
+        self.x += (self.move_x + self.momentum_x) * lag_scalar
+        self.y += (self.move_y + self.momentum_y) * lag_scalar
+        radius = self.collider.radius
+        if (self.impacted
```

Comment:
There is a "TODO" in level_one.py you can remove since the projectiles now remove themselves when they go off screen.  "# TODO: remove elements no longer on screen?"